scripts/egp27/createProposalUpdateGrants.ts

```diff
@@ -0,0 +1,236 @@
+import vaultInterface from "artifacts/contracts/vaults/UnfrozenVestingVault.sol/UnfrozenVestingVault.json";
+import { ethers, Signer } from "ethers";
+import hre from "hardhat";
+import { CoreVoting__factory, SimpleProxy__factory } from "typechain";
+
+import addressesJson from "src/addresses";
+import { DAY_IN_BLOCKS } from "src/constants";
+import { createCallHash } from "src/helpers/createCallHash";
+import {
+  AddGrant,
+  Grant,
+  ProposalArgs,
+  ProposalInfo,
+  ReduceGrant,
+} from "src/types";
+
+export async function createProposalUpdateGrants(
+  signer: Signer,
+  grants: Grant[],
+  unfrozenVaultAddress: string,
+  votingVaultAddresses: string[],
+  extraVaultDatas: string[]
+): Promise<ProposalInfo> {
+  console.log("grants", grants);
+  const provider = hre.ethers.getDefaultProvider();
+
+  const { coreVoting, timeLock, vestingVault } = addressesJson.addresses;
+  const vestingVaultProxy = SimpleProxy__factory.connect(vestingVault, signer);
+  const frozenVaultAddress = await vestingVaultProxy.proxyImplementation();
+
+  const proposalInfo = await createVestingGrantsUpgradeProposal(
+    signer,
+    provider,
+    coreVoting,
+    timeLock,
+    vestingVault,
+    unfrozenVaultAddress,
+    frozenVaultAddress,
+    votingVaultAddresses,
+    extraVaultDatas,
+    grants
+  );
+
+  return proposalInfo;
+}
+
+export async function createVestingGrantsUpgradeProposal(
+  signer: Signer,
+  provider: Provider,
+  coreVotingAddress: string,
+  timeLockAddress: string,
+  // vesting vault proxy address
+  vestingVaultAddress: string,
+  // upgraded vesting vault implementation address
+  unfrozenVaultAddress: string,
+  // original vesting vault implementation address
+  frozenVaultAddress: string,
+  // voting vaults to query vote power from to submit proposal
+  votingVaultAddresses: string[],
+  // extra data for voting vaults if necessary
+  extraVaultData: string[],
+  // grants to add/remove
+  grants: Grant[]
+): Promise<ProposalInfo> {
+  /********************************************************************************
+   * Set up a new proposal.  This proposal will perform 3 actions:
+   *   1. temporarily upgrade the VestingVault's implementation to one that can add AND remove grants
+   *   2. perform necessary updates to grants
+   *   3. reset the VestingVault back to the original implementation.
+   ********************************************************************************/
+
+  // step 1 is to update the vesting vault implementation address
+  const {
+    targets,
+    callDatas,
+    proposalHash,
+    targetsTimeLock,
+    calldatasTimeLock,
+    callHashTimelock,
+  } = await getUpdateGrantsProposalArgs(
+    provider,
+    grants,
+    unfrozenVaultAddress,
+    frozenVaultAddress,
+    vestingVaultAddress,
+    timeLockAddress
+  );
+
+  const ballot = 0; // 0 - YES, 1 - NO, 2 - ABSTAIN
+
+  // create the arguments to coreVoting.proposal()
```

Comment:
Nit: These comments are fairly misleading. No worries if this is just for you to remember, but it would be good to update these if we'll want other people to help maintain these scripts.

Reply:
Ok I removed this one and fixed a couple others.  I can do a full sweep of comments as well to spruce this up if you show me more nits.